corehq/apps/ota/views.py

```diff
@@ -61,15 +68,23 @@ def restore(request, domain, app_id=None):
     tags = [
         u'status_code:{}'.format(response.status_code),
     ]
+    env = settings.SERVER_ENVIRONMENT
+    if (env, domain) in settings.RESTORE_TIMING_DOMAINS:
+        tags.append(u'domain:{}'.format(domain))
     datadog_counter('commcare.restores.count', tags=tags)
     if timing_context is not None:
+        datadog_histogram(
+            'commcare.restores.total_time',
+            timing_context.duration,
+            tags=tags,
+        )
         for timer in timing_context.to_list(exclude_root=True):
-            # Only record leaf nodes so we can sum to get the total
-            if timer.is_leaf_node:
+            if timer.name in TIMED_RESTORE_SEGMENTS:
+                segment = TIMED_RESTORE_SEGMENTS[timer.name]
                 datadog_histogram(
                     'commcare.restores.timings',
                     timer.duration,
-                    tags=tags + [u'segment:{}'.format(timer.name)],
+                    tags=tags + [u'segment:{}'.format(segment)],
```

Comment:
Another idea I had was to add the segment to the metric name (e.g., `commcare.restore.timings.cases`) and remove the segment tag. This might make it easier to do some things in datadog. However, I'm not sure if that would make it impossible to make a stacked graph of all segments?